### ISSUE TITLE
Freezing trie taking > 1 hour to complete

### ISSUE THREAD

**Issue by ezzatron** [devongovett/unicode-trie#2]
I've been using `unicode-trie` over at [eloquent/precis-js](https://github.com/eloquent/precis-js), and it's been working great. However, I just recently decided to re-build my trie, and it took in excess of 1 hour to complete.

I don't remember it taking that long previously, but it has been a while since I've rebuilt, and this is on a reasonably fresh install of OSX El Capitan. I threw in some quick debug code, and tracked the issue down to `UnicodeTrieBuilder._compactData()`, but beyond that I don't know.

The process to reproduce is:

    git clone git@github.com:eloquent/precis-js.git
    cd precis-js
    npm install
    scripts/generate-data

Any ideas why it would be taking so long?

**Comment by devongovett** [comment]
Wow, turns out this was a really bad bug. I ported that code from C, and I guess I made some mistakes converting it. Found a few places where the wrong number of args were being passed to functions, causing the slowness, and causing the generated trie to be much bigger than it needed to be (block deduplication was broken).

Fixed it in ab0fd8194a520673d5efa40fb800b0835adf1307. On one of my projects, the output file was reduced from 12KB to 8KB after the change. Ran your code with the new version just released (v0.3.1), and generating the trie is super fast now.

One other note: calling `trie.freeze()` [here](https://github.com/eloquent/precis-js/blob/develop/scripts/generate-data.coffee#L50) is not necessary since `trie.toBuffer` does that for you.
